addon/components/hyper-table.js

```diff
@@ -156,6 +157,22 @@ export default Component.extend({
     );
   },
 
+  groupByClusteringKey(fields) {
+    const map = new Map();
+
+    fields.forEach((field) => {
+      const cluster = map.get(field.clustering_key);
+
+      if (!cluster) {
+        map.set(field.clustering_key, [field]);
+      } else {
+        cluster.push(field);
+      }
+    });
```

Comment:
IMO this can be simplify `(map[field.clustering_key] || (map[field.clustering_key] = [])).push(field) ` but maybe not clearer haha

Reply:
@Hegazy360  +1

Reply:
I tried this solution, I'm getting weird results
Found out that when assigning a `push` to a variable, it returns the `length` instead, I tried modifying it a bit but it feels like an overkill

Reply:
yes my bad it's more `(map[field.clustering_key] || (map[field.clustering_key] = [])).push(field)` :p

Reply:
should be good